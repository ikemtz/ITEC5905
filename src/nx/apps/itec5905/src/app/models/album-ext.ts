
import { IAlbum } from "../../models/artists-webapi";
export interface IAlbumExt extends IAlbum {
  image?: string | null;
}