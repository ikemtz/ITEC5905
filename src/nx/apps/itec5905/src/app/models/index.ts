export * from './album-ext';
export * from './album-upsert';
export * from './artist-ext';
export * from './artist-upsert';
export * from './song-ext';
export * from './song-upsert';
