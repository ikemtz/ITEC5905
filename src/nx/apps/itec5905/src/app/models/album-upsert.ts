import { IMediaUploadRequest } from "../../models/media-webapi";
import { IAlbumUpsertRequest, } from "../../models/artists-webapi";

export interface IAlbumUpsert {
  album: IAlbumUpsertRequest,
  picture?: IMediaUploadRequest,
}