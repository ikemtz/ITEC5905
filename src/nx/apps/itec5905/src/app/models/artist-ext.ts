
import { IArtist } from "../../models/artists-webapi";
export interface IArtistExt extends IArtist {
  image?: string | null;
}