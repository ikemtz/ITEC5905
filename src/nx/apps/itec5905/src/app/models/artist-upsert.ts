import { IMediaUploadRequest } from "../../models/media-webapi";
import { IArtistUpsertRequest } from "../../models/artists-webapi";

export interface IArtistUpsert {
  artist: IArtistUpsertRequest,
  picture?: IMediaUploadRequest,
}