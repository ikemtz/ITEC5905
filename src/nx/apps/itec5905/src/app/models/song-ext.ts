
import { ISong } from "../../models/artists-webapi";
export interface ISongExt extends ISong {
  image?: string | null;
}