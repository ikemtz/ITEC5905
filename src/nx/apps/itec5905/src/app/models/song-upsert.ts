import { ISongUpsertRequest } from "../../models/artists-webapi";
import { IMediaUploadRequest } from "../../models/media-webapi";

export interface ISongUpsert {
  song: ISongUpsertRequest,
  picture?: IMediaUploadRequest,
  audio?: IMediaUploadRequest,
}