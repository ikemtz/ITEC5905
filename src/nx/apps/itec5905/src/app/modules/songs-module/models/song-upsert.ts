import { ISongUpsertRequest } from "apps/itec5905/src/models/artists-webapi";
import { IMediaUploadRequest } from "apps/itec5905/src/models/media-webapi";

export interface ISongUpsert {
  song: ISongUpsertRequest,
  picture?: IMediaUploadRequest,
  audio?: IMediaUploadRequest,
}