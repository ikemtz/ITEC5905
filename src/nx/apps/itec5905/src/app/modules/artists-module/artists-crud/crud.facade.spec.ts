import { NgModule } from '@angular/core';
import { TestBed } from '@angular/core/testing';
import { HttpClient } from '@angular/common/http';
import { EffectsModule } from '@ngrx/effects';
import { StoreModule, Store } from '@ngrx/store';
import { readFirst } from 'imng-ngrx-utils/testing';
import {
  testAddSetAndClearCurrentEntity,
  testEditSetAndClearCurrentEntity,
  testSaveCurrentEntity,
  testUpdateCurrentEntity,
} from 'imng-kendo-data-entry/testing';
import { createODataPayload } from 'imng-kendo-odata';
import { of } from 'rxjs';

import { ArtistEffects } from '../+state/artist.effects';
import { artistsFeature } from '../+state/artist.reducer';
import { ArtistCrudFacade } from './crud.facade';
import { ArtistApiService } from './api.service';
import { environment } from '../../../../environments/environment';
import { createTestArtist } from '../../../../models/artists-webapi';

describe('ArtistCrudFacade', () => {
  let facade: ArtistCrudFacade;
  let store: Store;
  let httpClient: HttpClient;

  // eslint-disable-next-line @typescript-eslint/no-empty-function
  beforeEach(() => { }); //NOSONAR

  describe('used in NgModule', () => {
    beforeEach(() => {
      @NgModule({
        imports: [
          StoreModule.forFeature(artistsFeature),
          EffectsModule.forFeature([ArtistEffects]),
        ],
        providers: [
          ArtistCrudFacade,
          ArtistApiService,
          { provide: HttpClient, useValue: { get: jest.fn(() => of(createODataPayload([createTestArtist()]))) } },
        ],
      })
      class CustomFeatureModule { }

      @NgModule({
        imports: [
          StoreModule.forRoot({}, { runtimeChecks: environment.runtimeChecks }),
          EffectsModule.forRoot([]),
          CustomFeatureModule,
        ],
      })
      class RootModule { }
      TestBed.configureTestingModule({ imports: [RootModule] });

      // eslint-disable-next-line @typescript-eslint/no-unused-vars
      store = TestBed.inject(Store);
      facade = TestBed.inject(ArtistCrudFacade);
      httpClient = TestBed.inject(HttpClient);
    });

    test('clearCurrentEntity() should set currentArtist to null', async () => {
      let isNewActive = await readFirst(facade.isNewActive$);
      expect(isNewActive).toBeFalsy();

      facade.clearCurrentEntity();
      isNewActive = await readFirst(facade.isNewActive$);

      expect(isNewActive).toBeFalsy();
    });

    test('New Entity Set And Clear CurrentEntity', async () =>
      testAddSetAndClearCurrentEntity<ArtistCrudFacade>(facade));
    test('Existing Entity Set And Clear CurrentEntity', async () =>
      testEditSetAndClearCurrentEntity<ArtistCrudFacade>(facade));
    test('Save CurrentEntity', async () =>
      testSaveCurrentEntity<ArtistCrudFacade>(facade, httpClient));
    test('Update CurrentEntity', async () =>
      testUpdateCurrentEntity<ArtistCrudFacade>(facade, httpClient));

    test('should load Pictures', async () => {
      facade.loadPictures({});
      expect(httpClient.get).toBeCalledTimes(1);
      const result = await readFirst(facade.pictures$);
      expect(result.length).toBe(1);
    });
  });
});
