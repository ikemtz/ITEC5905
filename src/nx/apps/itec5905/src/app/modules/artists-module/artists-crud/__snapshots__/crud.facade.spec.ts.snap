// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistCrudFacade used in NgModule clearCurrentEntity() should set currentArtist to null 1`] = `
{
  "artists": {
    "currentArtist": undefined,
    "error": undefined,
    "gridData": {
      "data": [],
      "total": 0,
    },
    "gridODataState": {},
    "gridPagerSettings": false,
    "loading": true,
  },
}
`;
